### PR TITLE
Stop using jupyter-execute in release notes

### PR DESCRIPTION
**Pull request by Eric-Arellano**
### Summary

jupypter-execute has problems, as explained at https://github.com/Qiskit/qiskit-terra/issues/7661#issue-1137695552. One issue with `jupyter-execute` in release notes is it cannot handle deprecation changes well. For example, Terra made tweedledum optional; but due to old `jupyter-execute` blocks using those old features, https://github.com/Qiskit/qiskit/pull/1674 is failing.

For this reason, Terra has already removed Jupyter-execute from all its Reno release notes. But, those release notes have not been rebuilt to this meta repository. This PR ~applies those updates (technically, they're now mostly using `plot`, whereas this uses code blocks. That's fine because these are old release notes.)

### Details and comments

### CODE REVIEW

File: docs/release_notes.rst [hunk 1]
```diff
@@ -3042,7 +3042,7 @@ New Features
   :meth:`.QuantumCircuit.draw` this label will be printed at the top of the
   ``barrier``.
 
-  .. jupyter-execute::
+  .. code-block:: python
```
Comment (Eric-Arellano) [inline docs/release_notes.rst]: If we want to keep some `jupyter-execute` around, maybe a better approach is to only remove the blocks for Old™️ versions of Qiskit? For example, this block is from Terra 0.22 - so maybe it is actually Worth It to keep for now!

Whereas something from 2 years ago isn't worth the performance hit & deprecation issues

**Comment by Eric-Arellano** [comment]
This seems fishy. Docs took 37 minutes vs 27 minutes on the latest `main` build at https://github.com/Qiskit/qiskit/actions/runs/4226514358/jobs/7340033415. @mtreinish any idea why this might be? I even reran the docs job in case caching is in play.

**Review by HuangJunye — COMMENTED** [review]
The release note is getting very long time to load and build as noted in https://github.com/Qiskit/qiskit/issues/1653 and https://github.com/Qiskit/qiskit/issues/1504, though I am not entirely sure whether `jupyter-execute` is to be blamed for the long load time or just the fact that the release note is very long.

Nevertheless, this is one good step towards getting rid of `jupyter-execute`. I also agree that executable code for deprecated features are hard to handle. Perhaps we can put manual output inside the release note at the time of writing.

**Review by jakelishman — APPROVED** [review]
should be fine to change all of these - we did it on terra a little while back, when we completely removed the use of `jupyter-sphinx` from the repo.  We swapped to using matplotlib's `plot` directive for actual docs, but old release notes don't need to bother.

**Comment by Eric-Arellano** [comment]
> we did it on terra a little while back, when we completely removed the use of jupyter-sphinx from the repo. We swapped to using matplotlib's plot directive for actual docs, but old release notes don't need to bother.

That's what I was thinking - Terra's Reno notes don't use jupyter-execute anymore, but these weren't updated. And that's causing pain like needing Tweedledum to build the docs.

Thanks for the review!

**Comment by jakelishman** [comment]
yeah, they're "out of date" because there's no release-notes automation for the metapackage docs because reno isn't really designed for it.  We always just copy them in manually off a `reno report` as part of the metapackage release, so they get frozen in time

**Pull request merged**